src/renderer/components/FirmwareUpdateBanner.js

```diff
@@ -0,0 +1,57 @@
+// @flow
+
+import React from "react";
+import IconNano from "~/renderer/icons/NanoAltSmall";
+import { Trans } from "react-i18next";
+import { useSelector } from "react-redux";
+import { useLocation, useHistory } from "react-router-dom";
+import { latestFirmwareSelector } from "~/renderer/reducers/settings";
+import TopBanner, { FakeLink } from "~/renderer/components/TopBanner";
+import getCleanVersion from "~/renderer/screens/manager/FirmwareUpdate/getCleanVersion";
+
+const FirmwareUpdateBanner = ({ old, right }: { old?: boolean, right?: any }) => {
+  const history = useHistory();
+  const location = useLocation();
+  const latestFirmware = useSelector(latestFirmwareSelector);
+  const visibleFirmwareVersion =
+    process.env.DEBUG_FW_VERSION ||
+    (latestFirmware ? getCleanVersion(latestFirmware.final.name) : "");
+
+  const onClick = () => {
+    const urlParams = new URLSearchParams({
+      firmwareUpdate: "true",
+    });
+    const search = urlParams.toString();
+    history.push({
+      pathname: "manager",
+      search: `?${search}`,
+    });
+  };
+
+  const inManager = location.pathname === "/manager";
+
+  return !visibleFirmwareVersion ? null : right || !inManager ? (
+    <TopBanner
+      id={"fw-update-banner"}
+      content={{
+        Icon: IconNano,
+        message: (
+          <Trans
+            i18nKey={
+              old ? "manager.firmware.banner.old.warning" : "manager.firmware.banner.warning"
```

Comment:
--This message is just a suggestion and not a blocking comment--

For stuff like that, we can use `context` from the i18n library, the feature fit perfectly with this kind of behaviour:

Documentation: https://www.i18next.com/translation-function/context#context

---

JSON file
```json
{
      "warning": "...",
      "warning_old": "...",
} 
```

JS file
```jsx
<Trans i18nKey="manager.firmware.banner.warning" tOptions={{context: old ? "old" : null}} values={{ latestFirmware: visibleFirmwareVersion }} />
```

It's clearer form the translation file that both keys are paired.

Reply:
Feel free to commit this to the pr, looks clearer to me too